File: pydroid/src/main/java/com/pyamsoft/pydroid/util/DrawableUtil.java
/*
 * Copyright 2017 Peter Kenji Yamanaka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.pyamsoft.pydroid.util;

import android.content.Context;
import android.graphics.ColorFilter;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffColorFilter;
import android.graphics.drawable.Drawable;
import android.support.annotation.CheckResult;
import android.support.annotation.ColorInt;
import android.support.annotation.ColorRes;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import com.pyamsoft.pydroid.helper.Checker;

public final class DrawableUtil {

  private DrawableUtil() {
    throw new RuntimeException("No instances");
  }

  @NonNull @CheckResult public static ColorFilter colorFilter(final @ColorInt int color) {
    return new PorterDuffColorFilter(color, PorterDuff.Mode.SRC_IN);
  }

  @NonNull @CheckResult
  public static Drawable tintDrawableFromColor(@NonNull Drawable d, final @ColorInt int c) {
    d = Checker.checkNonNull(d).mutate();
    d.setColorFilter(colorFilter(c));
    return d;
  }

  @NonNull @CheckResult
  public static Drawable tintDrawableFromRes(final @NonNull Context c, final @NonNull Drawable d,
      final @ColorRes int cl) {
    final @ColorInt int i = ContextCompat.getColor(c, cl);
    return tintDrawableFromColor(d, i);
  }
}

File: pydroid-util/src/main/java/com/pyamsoft/pydroid/util/DrawableUtil.kt
/*
 *     Copyright (C) 2017 Peter Kenji Yamanaka
 *
 *     This program is free software; you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation; either version 2 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License along
 *     with this program; if not, write to the Free Software Foundation, Inc.,
 *     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package com.pyamsoft.pydroid.util

import android.content.Context
import android.graphics.ColorFilter
import android.graphics.PorterDuff
import android.graphics.PorterDuffColorFilter
import android.graphics.drawable.Drawable
import android.support.annotation.CheckResult
import android.support.annotation.ColorInt
import android.support.annotation.ColorRes
import android.support.v4.content.ContextCompat

object DrawableUtil {

    @JvmStatic
    @CheckResult
    fun colorFilter(@ColorInt color: Int): ColorFilter =
        PorterDuffColorFilter(color, PorterDuff.Mode.SRC_IN)

    @JvmStatic
    @CheckResult
    fun tintDrawableFromColor(d: Drawable, @ColorInt c: Int): Drawable {
        d.colorFilter = colorFilter(c)
        return d
    }

    @JvmStatic
    @CheckResult
    fun tintDrawableFromRes(
        c: Context, d: Drawable,
        @ColorRes cl: Int
    ): Drawable {
        @ColorInt val i: Int = ContextCompat.getColor(c, cl)
        return tintDrawableFromColor(d, i)
    }
}
